src/web_app/js/appcontroller.js

```diff
@@ -193,7 +193,14 @@ AppController.prototype.showRoomSelection_ = function() {
 AppController.prototype.finishCallSetup_ = function(roomId) {
   this.call_.start(roomId);
 
-  window.onbeforeunload = this.call_.hangup.bind(this.call_);
+  // Chrome apps can't use onbeforeunload.
+  if (!isChromeApp()) {
+    // Call hangup with async = false. Required to complete multiple
+    // clean up steps before page is closed.
+    window.onbeforeunload = function() {
+      this.call_.hangup(false);
+    }.bind(this);
+  }
```

Comment:
Can this change be combined with the other `if` statement below so there is just a single code path where we are considering non chrome app?